compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2024,9 +2471,12 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         let name = match res {
             LifetimeRes::Param { param, .. } => {
                 let p_name = ParamName::Plain(ident);
-                let param = self.resolver.get_remapped_def_id(param);
-
-                hir::LifetimeName::Param(param, p_name)
+                if self.in_scope_generics.contains(&param) {
+                    hir::LifetimeName::Static
+                } else {
+                    let param = self.resolver.get_remapped_def_id(param);
+                    hir::LifetimeName::Param(param, p_name)
+                }
```

Comment:
Please document this. From the naming `in_scope_generics` it is not obvious why these are remapped to `'static`